jwst/dq_init/dq_initialization.py

```diff
@@ -134,7 +134,7 @@ def check_dimensions(input_model):
         # Perform the same check for the input model groupdq array
         if input_model.groupdq.shape != input_shape:
             if input_model.groupdq.shape == (0, 0, 0, 0):
-                input_model.groupdq = np.zeros((input_shape)).astype('uint8')
+                input_model.groupdq = np.zeros((input_shape)).astype('uint32')
```

Comment:
This line is doing an initial population of the `groupdq` data array in the input data model if the code finds that the data array does not exist. And given that the `groupdq` is defined to be `uint8` in our datamodels schemas (see https://github.com/spacetelescope/stdatamodels/blob/main/src/stdatamodels/jwst/datamodels/schemas/ramp.schema.yaml#L29), I think this should not be changed. This is not just a random internal variable or array that's being used temporarily in the step. This is the official `groupdq` data array for the data model, which is defined to be `uint8`.

Reply:
So what happens if the code tries to set a high-valued bit for groupdq, such as 2**9=512?  Or by construction are all of the bits that groupdq might ever set less than 256?  (I.e., last possible flag would be 2**7=128 for CHARGELOSS?)

Reply:
> So what happens if the code tries to set a high-valued bit for groupdq, such as 2**9=512? Or by construction are all of the bits that groupdq might ever set less than 256? (I.e., last possible flag would be 2**7=128 for CHARGELOSS?)

If the code is implemented properly, which means always using pre-defined group-dependent flag values to set bits in the ``groupdq`` data array (e.g. dqflags.group['chargeloss']) it's impossible to ever have a value larger than the 8-bit limit assigned, because the ``dqflags.group`` definitions only include 8 possible settings.

Reply:
And because the ``groupdq`` data array is defined to be ``uint8``, you would run into serious trouble if anyone ever tried assigning a value that exceeds the 8-bit limit when the data model gets serialized to a FITS file. When writing to a file, the data model schema is strictly enforced, i.e. the ``groupdq`` data array is forced to be written with data type ``uint8``. So if you allowed larger values to be assigned to it somewhere in the code, it would then give very bad results when writing the array into the FITS file.